android/java/org/chromium/chrome/browser/searchwidget/SearchWidgetProviderActivity.java

```diff
@@ -7,13 +7,25 @@
 
 package org.chromium.chrome.browser.searchwidget;
 
+import android.app.SearchManager;
 import android.content.Intent;
 
+import org.chromium.base.IntentUtils;
 import org.chromium.chrome.browser.searchwidget.SearchActivity;
+import org.chromium.chrome.browser.ui.searchactivityutils.SearchActivityConstants;
 
 public class SearchWidgetProviderActivity extends SearchActivity {
     @Override
     public void onNewIntent(Intent intent) {
-        super.onNewIntent(intent);
+        Intent newIntent = new Intent();
+        if (IntentUtils.safeGetStringExtra(intent, SearchManager.QUERY) != null) {
+            newIntent.putExtra(
+                    "query", IntentUtils.safeGetStringExtra(intent, SearchManager.QUERY));
```

Comment:
non-blocking: for clarity, let's use `SearchManager.QUERY` as the first parameter. It resolves to "query", but could be changed without notice.